Frontend/src/Base/BaseImageUpload/BaseImageUpload.jsx

```diff
@@ -0,0 +1,102 @@
+import { useRef, useState } from 'react';
+import BaseButton from '../BaseButton';
+import { useTranslation } from 'react-i18next';
+import UploadedImagesDisplay from './UploadedImagesDisplay';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectAddPlace, addPlaceActions } from '../../Redux/addPlaceSlice';
+
+const BaseImageUpload = ({ fileSize }) => {
+  const dispatch = useDispatch();
+  const images = useSelector(selectAddPlace).images;
+  const [errorMsg, setErrorMsg] = useState(null);
+  const [showError, setShowError] = useState(null);
+  const fileInputRef = useRef(null);
+  const { t } = useTranslation();
+  const MAX_FILE_SIZE = fileSize * 1024 * 1024;
+  const MAX_FILES = 3;
+
+  const handleClick = () => {
+    fileInputRef.current.click();
+  };
+
+  const validateImagesQuantity = (uploadedImagesLength) => {
+    const allFilesLenght = uploadedImagesLength + images.length;
+
+    if (uploadedImagesLength > MAX_FILES || allFilesLenght > MAX_FILES) {
+      setErrorMsg(`${t('common.max_files_number')} ${MAX_FILES}`);
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const validateDuplicates = (uploadedImage) => {
+    const duplicateFile = images.find((image) => image.name === uploadedImage.name);
+    if (duplicateFile) {
+      setErrorMsg(`${t('common.already_uploaded')} ${uploadedImage.name}`);
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const validateImageType = (uploadedImage) => {
+    if (!uploadedImage.type.startsWith('image/')) {
+      setErrorMsg(t('common.only_images'));
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const validateImageSize = (uploadedImage) => {
+    if (uploadedImage.size >= MAX_FILE_SIZE) {
+      setErrorMsg(`${t('common.file_too_large')} ${uploadedImage.name}`);
+      return false;
+    } else {
+      return true;
```

Comment:
Po co zwracasz true możesz zrobić to bez tego i usunąć tego else a w if-e zrobić tak :
`if (!validateImageSize(image)) do_something ;`

Reply:
Tylko wtedy polecam inną nawzę np. wrong_size_cheker albo wg zmienić logikę i sprawdzać czy jest okej i zwracać true

Reply:
Co do if-a to musi zwracać true or false ale mogę zredukować trochę kod, przykład poniżej:
`const validateImageSize = (uploadedImage) => {
  const isImageTooLarge = uploadedImage.size >= MAX_FILE_SIZE;
  isImageTooLarge && setErrorMsg(${t('common.file_too_large')} ${uploadedImage.name});
  return !isImageTooLarge;
};`

Reply:
```
const validateAA = (uploadedImagesLength) => {
    if (true) {
      return true;
    }
    return false 
  };

  const validate_AA = (uploadedImage) => {
    if (true) {
      return true;
    }
    return false 
  };

  const validateA_A = (uploadedImage) => {
    if (true) {
      return true;
    }
    return false 
  };

  const validateA_B = (uploadedImage) => {
    if (true) {
      return true;
    }
    return false 
  };

const cheker = ()=>{
  var a =null
  var l =[validateAA,validateA_A,validateA_B,validate_AA]
  var is_valid = l.map(validator => validator(a))  
  if (is_valid.every(result=>result)){
        do_something()
  }
};

Reply:
Tutaj taki wzór masz